test/integration/model/json.test.js

```diff
@@ -508,4 +508,192 @@ describe(Support.getTestDialectTeaser('Model'), () => {
+                first: 'Homer',
+                last: 'Simpson'
+              },
+              employment: 'Nuclear Safety Inspector'
+            });
+          });
+        });
+      });
+
+      it('should query an instance with JSONB data', function () {
+        return this.Event.create({
+          data: {
+            name: {
+              first: 'Homer',
+              last: 'Simpson'
+            },
+            employment: 'Nuclear Safety Inspector'
+          }
+        }).then(() => {
+          return this.Event.findAll({
+            where: {
+              'data.name.first': 'Homer'
+            }
+          }).then(events => {
+            const event = events[0];
+
+            expect(event.get('data')).to.eql({
+              name: {
+                first: 'Homer',
+                last: 'Simpson'
+              },
+              employment: 'Nuclear Safety Inspector'
+            });
+          });
+        });
+      });
+
+      it('should query an instance with JSONB data and order', function () {
+        return this.Event.create({
+          data: {
+            name: {
+              first: 'Homer',
+              last: 'Simpson'
+            },
+            employment: 'Nuclear Safety Inspector'
+          }
+        }).then(() => {
+          return Promise.join(
+            this.Event.create({
+              data: {
+                name: {
+                  first: 'Marge',
+                  last: 'Simpson'
+                },
+                employment: 'Housewife'
+              }
+            }),
+            this.Event.create({
+              data: {
+                name: {
+                  first: 'Bart',
+                  last: 'Simpson'
+                },
+                employment: 'None'
+              }
+            })
+          );
+        }).then(() => {
+          return this.Event.findAll({
+            where: {
+              data: {
+                name: {
+                  last: 'Simpson'
+                }
+              }
+            },
+            order: [
+              ['data.name.first']
+            ]
+          }).then(events => {
+            expect(events.length).to.equal(3);
+
+            expect(events[0].get('data')).to.eql({
+              name: {
+                first: 'Bart',
+                last: 'Simpson'
+              },
+              employment: 'None'
+            });
+
+            expect(events[1].get('data')).to.eql({
+              name: {
+                first: 'Homer',
+                last: 'Simpson'
+              },
+              employment: 'Nuclear Safety Inspector'
+            });
+
+            expect(events[2].get('data')).to.eql({
+              name: {
+                first: 'Marge',
+                last: 'Simpson'
+              },
+              employment: 'Housewife'
+            });
+          });
+        });
+      });
+
+      it('should query an instance with JSONB data and order while trying to inject', function () {
```

Comment:
Move this one to find section as well https://github.com/mkaufmaner/sequelize/blob/b5d77ecedbe96d47d053cb9f11e10cdf0f84d0ae/test/integration/model/json.test.js#L209